src/setup.jl

```diff
@@ -232,12 +232,12 @@ function locate_JuliaInterface_so(sysinfo::Dict{String, String})
     jll_hash = GitTools.tree_hash(joinpath(jll, "src"))
     bundled = joinpath(@__DIR__, "..", "pkg", "JuliaInterface")
     bundled_hash = GitTools.tree_hash(joinpath(bundled, "src"))
-    if jll_hash == bundled_hash
+    if jll_hash == bundled_hash && get(ENV, "FORCE_JULIAINTERFACE_COMPILATION", "false") != "true"
```

Comment:
Usually I'd just check for the presence of such a variable and not care about the content. But since this is not intended for user consumption, I am fine either way.